scripts/write_DAC_values.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+import argparse
+
+parser = argparse.ArgumentParser(description='Arguments to supply to write_DAC_values.py')
+
+parser.add_argument('OH', type=str, help="OH number", metavar='OH')
+parser.add_argument('nominalDacFileList', type=str, help="Name of a file containing a list of register names and NominalDACValues.txt files. Format:register_name1 <space> /path/to/nominal/DAC/file1 <newline> register_name2 <space> /path/to/nominal/DAC/file2 <newline> ... ", metavar='nominalDacFileList')
+parser.add_argument('--dry_run', dest='dry_run', action='store_true', help="If this flag is set the script will not overwrite the vfat3 config files.")
+
+args = parser.parse_args()
+
+#values copied from https://github.com/cms-gem-daq-project/cmsgemos/blob/generic-amc-RPC-v3-short-term/gempython/tools/amc_user_functions_xhal.py#L14-L43
+max_DAC_values = {
+            "BIAS_PRE_I_BIT": 0xff,
+            "BIAS_PRE_I_BLCC": 0x3f,
+            "BIAS_SH_I_BFCAS": 0xff,
+            "BIAS_SH_I_BDIFF": 0xff,
+            "BIAS_SD_I_BDIFF": 0xff,
+            "BIAS_SD_I_BFCAS": 0xff,
+            "BIAS_SD_I_BSF": 0x3f,
+            "BIAS_CFD_DAC_1": 0x3f,
+            "BIAS_CFD_DAC_2": 0x3f,
+            "HYST": 0x3f,
+            "THR_ARM_DAC": 0xff,
+            "THR_ZCC_DAC": 0xff,
+            "BIAS_PRE_VREF": 0xff,
+            "THR_ARM_DAC": 0xff,
+            "THR_ZCC_DAC": 0xff,
+            "ADC_VREF": 0x3
+            }
+
+nominal_DAC_value_files = {}
+
+for line in open(args.nominalDacFileList):
+    if line[0] == "#":
+        continue
+    line = line.strip(' ').strip('\n')
+    first = line.split(' ')[0].strip(' ')
+    second = line.split(' ')[1].strip(' ')
+
+    if first not in max_DAC_values.keys():
+        print('Warning: no maximum value found for register: '+first)
+    
+    nominal_DAC_value_files[first] = second
+
+for reg in nominal_DAC_value_files.keys():
+    print "setting the register "+str(reg)+ " using the file:"+nominal_DAC_value_files[reg]
+    fname=nominal_DAC_value_files[reg]
+    f = open(nominal_DAC_value_files[reg])
+    for line in f:
+        vfat=int(line.split('\t')[0])
+        value=int(line.split('\t')[1])
+
+        if value < 0:
+            value = 0
+
+        if vfat not in [11,12]:     
+            if reg in max_DAC_values.keys() and value > max_DAC_values[reg]:
+                print str(vfat) +": "+ str(value) +" --> " + str(max_DAC_values[reg])
+                value = max_DAC_values[reg]
```

Comment:
Why isn't this done in [anaDACScan.py](https://github.com/cms-gem-daq-project/gem-plotting-tools/blob/develop/anaDACScan.py) where the `max_DAC_values` dict could be imported from `amc_user_functions_xhal.py`?

Seems like all this script should do is to take an input file and write whatever values are in that file to the CTP7 configuration files.  

Having this check on max value here duplicates a number of hard coded parameters which makes the overall framework harder to maintain in the long run.

This should be removed from here, and the check that is attempted should be done in the analysis routine itself (PR required).

Reply:
ok, I will remove that from here, and make a PR for gem-plotting-tools

Reply:
here https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/162 is the requested PR for gem-plotting-tools